miner/worker.go

```diff
@@ -334,17 +340,28 @@ func (miner *Miner) prepareWork(genParams *generateParams, witness bool) (*envir
 		return nil, err
 	}
 	env.noTxs = genParams.noTxs
-	if evicted := miner.feeCurrencyBlocklist.Evict(parent); len(evicted) > 0 {
-		log.Warn(
-			"Evicted temporarily blocked fee-currencies from local block-list",
-			"evicted-fee-currencies", evicted,
-			"eviction-timeout-seconds", EvictionTimeoutSeconds,
+	// don't do anything blocklist related when we are:
+	// - deriving from L1
+	// - building the pending block
+	// because this could poison the blocklist or prevent
+	// derivation from executing consensus blocks when the blocklist
+	// is poisoned.
+	env.blocklistsDisabled = env.noTxs || genParams.isPending
```

Comment:
So when we have noTxs==true in the payload it means don't use the txpool, meaning we are deriving rather than constructing a block.

When we have isPending==true that means that we are constructing the pending block, so not deriving.

So I think you need the following:
```suggestion
	env.blocklistsDisabled = env.noTxs || !genParams.isPending
```

Reply:
I also think it's unneccesarry to have the pending condition if you are going to build the pending block you must have noTxs==false.

Reply:
it was intended as committed - basically meaning `!env.blocklistsDisabled == "we building sequencing payloads"` and thus we should disable all blocklist related stuff when we are either deriving or calculating the pending block, since this means we are not sequencing

Reply:
Oh ok, I was thinking we'd allow the blocklist to be active during the pending block calculation 🤔

Reply:
I think it makes sense to limit it just to sequencing though. Assuming there's not some system dependency on the pending block being roughly correct.